src/service/validation/validation.marker.service.spec.ts

```diff
@@ -0,0 +1,210 @@
+    const authToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.t-IDcSemACt8x4iTMCda8Yhe3iZaWbvV5XKSTbuAn0M';
+
+    TestBed.configureTestingModule({
+      imports: [HttpModule],
+      providers: [
+        { provide: XHRBackend, useClass: MockBackend},
+        { provide: AuthConfig, useValue: new AuthConfig({tokenGetter: () => authToken}) },
+        { provide: XtextValidationMarkerServiceConfig, useValue: serviceConfig },
+        { provide: ValidationMarkerService, useClass: XtextNaiveValidationMarkerService},
+        AuthHttp
+      ]
+    });
+  });
+
+  it('retrieves markers for single file', async(inject([XHRBackend, ValidationMarkerService],
+    (backend: MockBackend, validationMarkerService: ValidationMarkerService) => {
+    // given
+    const sampleFile: WorkspaceElement = { path: 'sample/path/file.txt', name: 'file.txt', children: [], type: ElementType.File };
+    backend.connections.subscribe((connection: MockConnection) => {
+      connection.mockRespond(new Response(new ResponseOptions({
+        status: 200,
+        body: sampleResponseBody
+      })));
+    })
+
+    // when
+    validationMarkerService.getAllMarkerSummaries(sampleFile).then((summaries: ValidationSummary[]) => {
+
+    // then
+    expect(summaries.length).toEqual(1);
+    expect(summaries[0].path).toEqual(sampleFile.path);
+    expect(summaries[0].errors).toEqual(3);
+    expect(summaries[0].warnings).toEqual(2);
+    expect(summaries[0].infos).toEqual(6);
+  });
+  })));
+
+  it('retrieves markers for multiple (nested) files and folders', async(inject([XHRBackend, ValidationMarkerService],
+    (backend: MockBackend, validationMarkerService: ValidationMarkerService) => {
+    // given
+    backend.connections.subscribe((connection: MockConnection) => {
+      connection.mockRespond(new Response(new ResponseOptions({
+        status: 200,
+        body: sampleResponseBody
+      })));
+    })
+
+    // when
+    validationMarkerService.getAllMarkerSummaries(root).then((summaries: ValidationSummary[]) => {
+
+    // then
+    expect(summaries).toEqual(expectedValidationMarkersForSampleResponse);
+    console.log(JSON.stringify(summaries));
+  });
+  })));
+
+  it('handles errors in responses gracefully', async(inject([XHRBackend, ValidationMarkerService],
+    (backend: MockBackend, validationMarkerService: ValidationMarkerService) => {
+    // given
+    let connectionCounter = 0;
+    backend.connections.subscribe((connection: MockConnection) => {
+      if (connectionCounter++ % 2) {
+        connection.mockRespond(new Response(new ResponseOptions({
+          status: 200,
+          body: sampleResponseBody
+        })));
+      } else {
+        connection.mockError(new Error('Error while requesting validation markers'));
+      }
+    })
+
+    // when
+    validationMarkerService.getAllMarkerSummaries(root).then((summaries: ValidationSummary[]) => {
+
+    // then
+    expect(summaries.sort()).toEqual(expectedValdationMarkersWithErrors.sort());
```

Comment:
I'm missing some expectation that shows that errors and normal responses were actually used/triggered

Reply:
This is kind of contained in `expectedValdationMarkersWithErrors`, i.e. the expectation is that those validation markers that were received are correctly handled and summed up, while the markers that would have been received, but for which an error was received instead, are missing.

I'll add an expectation to check whether the total number of requests matches the number of workspace elements; that should communicate more clearly that both errors and regular responses are being handled – right?